### PR TITLE
Update parallel_test.py to support Ray RLLib MultiAgentEnv

### PR DESCRIPTION
**Pull request by chrisyeh96**
# Description

I updated `parallel_test.py` so that it only checks the truncated/terminated status of agents in `live_agents` (which was created from `self.agents`). This allows `parallel_test.py` to not fail on an environment that tries to implement RLLib's `MultiAgentEnv` API, which requires the `terminated` and `truncated` dicts to include a special `'__all__'` key.

See https://github.com/ray-project/ray/blob/3e8a1dc6064225a5039a171b2dc5afb0beb7f0f8/rllib/env/multi_agent_env.py#L106C14-L106C21

With this change, an environment that inherits from both `pettingzoo.ParallelEnv` and `ray.rllib.env.MultiAgentEnv` will pass the PettingZoo `parallel_api_test()` checker:

```python
from ray.rllib.env import MultiAgentEnv
from sustaingym.envs.cogen import CogenEnv

class CustomMultiAgentEnv(ParallelEnv, MultiAgentEnv):
    ...
```

I do not believe that this pull request changes `parallel_test.py` in any other meaningful way.

# Tests

I tried following the instructions in [CONTRIBUTING.md](https://github.com/Farama-Foundation/PettingZoo/blob/master/CONTRIBUTING.md) but was unable to get `pytest` to work. See #1094.

### CODE REVIEW

**Comment by elliottower** [comment]
Oh I didn’t know about this about RLlib, thanks for the contribution though this is very useful.

File: pettingzoo/test/parallel_test.py
Comment (elliottower) [inline pettingzoo/test/parallel_test.py:100]: This x[0] code is confusing, can you explain it and do some testing to make sure you’re understanding it correctly? It looks like it says (terminated false, either true) basically.

Comment (jjshoots) [reply pettingzoo/test/parallel_test.py:100]: Don't think that's necessary, the change does the same thing, may be a bit faster, and is easier to read as well.

**Comment by elliottower** [comment]
Talked with jet a bit about the details (second pair of eyes) and looks good, only thing is it looks like the tests are failing now

**Comment by elliottower** [comment]
Ahh okay I think the reason for two sets was this error: 

RuntimeError: Set changed size during iteration

So that part should be reverted but the rest looks good

**Comment by elliottower** [comment]
Oh I’m confusing this for your other PR I think? Anyways if you could look into the tests and try to fix this that would be great.

**Comment by chrisyeh96** [comment]
> Ahh okay I think the reason for two sets was this error:
> 
> RuntimeError: Set changed size during iteration
> 
> So that part should be reverted but the rest looks good


Ah, my bad. I accidentally iterated through the `live_agents` set while making changes to it. I've fixed this in a new commit.

File: pettingzoo/test/parallel_test.py
Comment (jjshoots) [inline pettingzoo/test/parallel_test.py:118]: Could we instead do something like:
```python
new_live_agents = []
for agent in live_agents:
  if terminated[agent] or truncated[agent]:
    break
  new_live_agents.append(agent)
live_agents = set(new_live_agents)
```

This would make it easier to read and for something like this to not happen in the future should someone else edit it.
Thanks!

**Review by jjshoots — CHANGES_REQUESTED** [review]
See previous comment.

File: pettingzoo/test/parallel_test.py
Comment (chrisyeh96) [reply pettingzoo/test/parallel_test.py]: I've rewritten it to completely eliminate this problem. Let me know what you think.

Comment (jjshoots) [reply pettingzoo/test/parallel_test.py]: Awesome, thanks for the help!

**Review by jjshoots — APPROVED** [review]
Just needs precommit, otherwise LGTM. Thanks for the contribution!

**Comment by chrisyeh96** [comment]
@jjshoots Is there anything I need to do on my end? I can't tell why pre-commit is failing.

**Comment by jjshoots** [comment]
@chrisyeh96 you need to do black on parallel_test.py [trace here](https://github.com/Farama-Foundation/PettingZoo/actions/runs/6154237502/job/16699287555?pr=1096#step:7:65)

**Comment by chrisyeh96** [comment]
@jjshoots Done.

**Comment by jjshoots** [comment]
@elliottower LGTM.

**Review by elliottower — APPROVED** [review]

**Pull request merged**
